account_banking_sepa_direct_debit/models/account_payment_order.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import _, exceptions, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = "account.payment.order"
+
+    def generate_payment_file(self):
+        """Creates the SEPA Direct Debit file. That's the important code !"""
+        self.ensure_one()
+        if self.payment_method_id.code != "sepa_direct_debit":
+            return super().generate_payment_file()
+        pain_flavor = self.payment_method_id.pain_version
+        # We use pain_flavor.startswith('pain.008.001.xx')
+        # to support country-specific extensions such as
+        # pain.008.001.02.ch.01 (cf l10n_ch_sepa)
+        if pain_flavor.startswith("pain.008.001.02"):
+            bic_xml_tag = "BIC"
+            name_maxsize = 70
+            root_xml_tag = "CstmrDrctDbtInitn"
+        elif pain_flavor.startswith("pain.008.003.02"):
+            bic_xml_tag = "BIC"
+            name_maxsize = 70
+            root_xml_tag = "CstmrDrctDbtInitn"
+        elif pain_flavor.startswith("pain.008.001.03"):
+            bic_xml_tag = "BICFI"
+            name_maxsize = 140
+            root_xml_tag = "CstmrDrctDbtInitn"
+        elif pain_flavor.startswith("pain.008.001.04"):
+            bic_xml_tag = "BICFI"
+            name_maxsize = 140
+            root_xml_tag = "CstmrDrctDbtInitn"
+        else:
+            raise UserError(
+                _(
+                    "Payment Type Code '%s' is not supported. The only "
+                    "Payment Type Code supported for SEPA Direct Debit are "
+                    "'pain.008.001.02', 'pain.008.001.03' and "
+                    "'pain.008.001.04'."
+                )
+                % pain_flavor
+            )
+        pay_method = self.payment_mode_id.payment_method_id
+        xsd_file = pay_method.get_xsd_file_path()
+        gen_args = {
+            "bic_xml_tag": bic_xml_tag,
+            "name_maxsize": name_maxsize,
+            "convert_to_ascii": pay_method.convert_to_ascii,
+            "payment_method": "DD",
+            "file_prefix": "sdd_",
+            "pain_flavor": pain_flavor,
+            "pain_xsd_file": xsd_file,
+        }
+        nsmap = self.generate_pain_nsmap()
+        attrib = self.generate_pain_attrib()
+        xml_root = etree.Element("Document", nsmap=nsmap, attrib=attrib)
+        pain_root = etree.SubElement(xml_root, root_xml_tag)
+        # A. Group header
+        (
+            group_header,
+            nb_of_transactions_a,
+            control_sum_a,
+        ) = self.generate_group_header_block(pain_root, gen_args)
+        transactions_count_a = 0
+        amount_control_sum_a = 0.0
+        lines_per_group = {}
+        # key = (requested_date, priority, sequence type)
+        # value = list of lines as objects
+        for line in self.bank_line_ids:
```

Comment:
This field does not exist due to the fact that the module bank.payment.line was deleted. Please, change it